src/combinatorics/young_diagram.lean

```diff
@@ -123,4 +128,128 @@ end distrib_lattice
 /-- Cardinality of a Young diagram -/
 @[reducible] protected def card (μ : young_diagram) : ℕ := μ.cells.card
 
+section transpose
+/-- The `transpose` of a Young diagram is obtained by swapping i's with j's. --/
+
+def transpose (μ : young_diagram) : young_diagram :=
+{ cells :=  (equiv.prod_comm _ _).finset_congr μ.cells,
+  is_lower_set := λ _ _ h, by {
+    simp only [finset.mem_coe, equiv.finset_congr_apply, finset.mem_map_equiv],
+    intro hcell, apply μ.is_lower_set _ hcell, simp [h], } }
+
+@[simp] lemma mem_transpose {μ : young_diagram} {c : ℕ × ℕ} : c ∈ μ.transpose ↔ c.swap ∈ μ :=
+by { change c ∈ μ.cells.map _ ↔ _, rw finset.mem_map_equiv, refl, }
+
+@[simp] lemma transpose_eq_iff {μ ν : young_diagram} : μ.transpose = ν ↔ μ = ν.transpose :=
+by { split; { rintro rfl, ext, simp } }
+
+@[simp] lemma transpose_transpose (μ : young_diagram) : μ.transpose.transpose = μ :=
+by rw transpose_eq_iff
+
+-- This is effectively both directions of the iff statement below.
+protected lemma le_of_transpose_le {μ ν : young_diagram} (h_le : μ.transpose ≤ ν) :
+  μ ≤ ν.transpose :=
+λ c hc, by { simp only [mem_transpose], apply h_le, simpa }
+
+@[simp] lemma transpose_le_iff {μ ν : young_diagram} : μ.transpose ≤ ν ↔ μ ≤ ν.transpose :=
```

Comment:
Again, for simp normal form it would be better to have `μ.transpose ≤ ν.transpose ↔ μ ≤ ν`.

I think you should have this too:
```
@[mono]
protected lemma transpose_mono {μ ν : young_diagram} (h_le : μ ≤ ν) : μ.transpose ≤ ν.transpose := ...
```

Reply:
Done.